src/Common/SurfaceFluxes/SurfaceFluxes.jl

```diff
@@ -37,366 +31,296 @@
   pages={652--657},
   year={1990}
 }
-
 """
 module SurfaceFluxes
 
 using RootSolvers
+using NLsolve
+
 using ..Thermodynamics
-using CLIMAParameters
+using DocStringExtensions
+using CLIMAParameters: AbstractEarthParameterSet
 using CLIMAParameters.Planet: molmass_ratio, grav
+using CLIMAParameters.SubgridScale: von_karman_const
 
-"""
-    compute_buoyancy_flux(param_set, shf, lhf, T_b, q, α_0)
-
-Computes buoyancy flux given
- - `shf` sensible heat flux
- - `lhf` latent heat flux
- - `T_b` surface boundary temperature
- - `q` phase partition (see [`PhasePartition`](@ref))
- - `α_0` specific volume
-"""
-function compute_buoyancy_flux(
-    param_set::AbstractParameterSet,
-    shf,
-    lhf,
-    T_b,
-    q,
-    α_0,
-)
-    FT = typeof(shf)
-    _molmass_ratio::FT = molmass_ratio(param_set)
-    _grav::FT = grav(param_set)
-    cp_ = cp_m(param_set, q)
-    lv = latent_heat_vapor(param_set, T_b)
-    temp1 = (_molmass_ratio - 1)
-    temp2 = (shf + temp1 * cp_ * T_b * lhf / lv)
-    return (_grav * α_0 / cp_ / T_b * temp2)
-end
-
-module Byun1990
+abstract type SurfaceFluxesModel end
 
-using RootSolvers
-using ...Thermodynamics
-using CLIMAParameters
-using CLIMAParameters.SubgridScale: von_karman_const
+struct Momentum end
+struct Heat end
 
-""" Computes ψ_m for stable case. See Eq. 12 """
-ψ_m_stable(ζ, ζ_0, β_m) = -β_m * (ζ - ζ_0)
+export surface_conditions, 
+    compute_R_z0,
+    compute_exchange_coefficients,
+    compute_f_h,
+    compute_f_m,
+    compute_Psi,
+    compute_psi,
+    Momentum,
+    Heat
 
-""" Computes ψ_h for stable case. See Eq. 13 """
-ψ_h_stable(ζ, ζ_0, β_h) = -β_h * (ζ - ζ_0)
+"""
+    SurfaceFluxConditions{FT}
 
-""" Computes ψ_m for unstable case. See Eq. 14 """
-function ψ_m_unstable(ζ, ζ_0, γ_m)
-    x(ζ) = sqrt(sqrt(1 - γ_m * ζ))
-    temp(ζ, ζ_0) = log1p((ζ - ζ_0) / (1 + ζ_0))  # log((1 + ζ)/(1 + ζ_0))
-    ψ_m = (
-        2 * temp(x(ζ), x(ζ_0)) + temp(x(ζ)^2, x(ζ_0)^2) - 2 * atan(x(ζ)) +
-        2 * atan(x(ζ_0))
-    )
-    return ψ_m
-end
+Surface flux conditions, returned from `surface_conditions`.
 
-""" Computes ψ_h for unstable case. See Eq. 15 """
-function ψ_h_unstable(ζ, ζ_0, γ_h)
-    y(ζ) = sqrt(1 - γ_h * ζ)
-    ψ_h = 2 * log1p((y(ζ) - y(ζ_0)) / (1 + y(ζ_0))) # log((1 + y(ζ))/(1 + y(ζ_0)))
-    return ψ_h
-end
+# Fields
 
+$(DocStringExtensions.FIELDS)
 """
-    monin_obukhov_len(param_set, u, flux)
+struct SurfaceFluxConditions{FT}
+    L_MO::FT
+    pottemp_flux_star::FT
```

Comment:
We should replace `pottemp_flux` by `dry_static_energy_flux` throughout. This is more consistent with our energetics, takes buoyancy of water vapor into account, and avoids unnecessary transformations of variables. (The relevant flux is Eq. (5.8) in the atmosphere design document, with boundary flux given by (6.8). We can neglect the kinetic energy in (5.8).)

Reply:
Fixed.